libraries-4/src/main/java/com/baeldung/jsoup/JsoupTableParser.java

```diff
@@ -0,0 +1,131 @@
+                .getResource(filePath)
+                .getFile());
+            doc = Jsoup.parse(input, "UTF-8", "");
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+        return doc;
+    }
+
+    // tableOrder - nth table in the document
+    public List<Map<String, String>> parseTable(Document doc, int tableOrder) {
+        Element table = doc.select("table")
+            .get(tableOrder);
+        Element tbody = table.select("tbody")
+            .get(0);
+        Elements dataRows = tbody.select("tr");
+        Elements headerRow = table.select("tr")
+            .get(0)
+            .select("th,td");
+
+        List<String> headers = new ArrayList<String>();
+        for (Element header : headerRow) {
+            headers.add(header.text());
+        }
+
+        List<Map<String, String>> parsedDataRows = new ArrayList<Map<String, String>>();
+        for (int row = 0; row < dataRows.size(); row++) {
+            Elements colVals = dataRows.get(row)
+                .select("th,td");
+
+            int colCount = 0;
+            Map<String, String> dataRow = new HashMap<String, String>();
+            for (Element colVal : colVals) {
+                dataRow.put(headers.get(colCount++), colVal.text());
+            }
+            parsedDataRows.add(dataRow);
+        }
+        return parsedDataRows;
+    }
+
+    public void updateTableData(Document doc, int tableOrder, String updateValue) {
+        Element table = doc.select("table")
+            .get(tableOrder);
+        Element tbody = table.select("tbody")
+            .get(0);
+        Elements dataRows = tbody.select("tr");
+
+        for (int row = 0; row < dataRows.size(); row++) {
+            Elements colVals = dataRows.get(row)
+                .select("th,td");
+
+            for (int colCount = 0; colCount < colVals.size(); colCount++) {
+                colVals.get(colCount)
+                    .text(updateValue);
+            }
+        }
+    }
+
+    public void addRowToTable(Document doc, int tableOrder) {
+        Element table = doc.select("table")
+            .get(tableOrder);
+        Element tbody = table.select("tbody")
+            .get(0);
+
+        Elements rows = table.select("tr");
+        Elements headerCols = rows.get(0)
+            .select("th,td");
+        int numCols = headerCols.size();
+
+        Elements colVals = new Elements(numCols);
+        for (int colCount = 0; colCount < numCols; colCount++) {
+            Element colVal = new Element("td");
+            colVal.text("11");
+            colVals.add(colVal);
+        }
+        Elements dataRows = tbody.select("tr");
+        Element newDataRow = new Element("tr");
+        newDataRow.appendChildren(colVals);
+        dataRows.add(newDataRow);
+        tbody.html(dataRows.toString());
+    }
+
+    public void deleteRowFromTable(Document doc, int tableOrder) {
```

Comment:
Shouldn't we name it deleteLastRowFromTable? or maybe re-think implementation to offer user which row to delete

Reply:
Changed method implementations, added rowNum parameter. So row will be deleted by rowIndex.